src/strategies/offer_forwarder/abstract/Forwarder.sol

```diff
@@ -237,10 +237,16 @@ abstract contract Forwarder is IForwarder, MangroveOffer {
     }
   }
 
-  ///@inheritdoc IOfferLogic
-  function retractOffer(IERC20 outbound_tkn, IERC20 inbound_tkn, uint offerId, bool deprovision)
-    public
-    override
+  ///@notice Retracts an offer from an Offer List of Mangrove.
+  ///@param outbound_tkn the outbound token of the offer list.
+  ///@param inbound_tkn the inbound token of the offer list.
+  ///@param offerId the identifier of the offer in the (`outbound_tkn`,`inbound_tkn`) offer list
+  ///@param deprovision if set to `true` if offer owner wishes to redeem the offer's provision.
+  ///@return freeWei the amount of native tokens (in WEI) that have been retrieved by retracting the offer.
+  ///@dev An offer that is retracted without `deprovision` is retracted from the offer list, but still has its provisions locked by Mangrove.
+  ///@dev Calling this function, with the `deprovision` flag, on an offer that is already retracted must be used to retrieve the locked provisions.
+  function _retractOffer(IERC20 outbound_tkn, IERC20 inbound_tkn, uint offerId, bool deprovision)
+    internal
     mgvOrOwner(outbound_tkn, inbound_tkn, offerId)
```

Comment:
I'd remove the `mgvOrOwner` from this internal function and adds it to public wrappers (including MangroveOrder). This is what we did for `_updateOffer`. This is because you might want a public wrapper that is `onlyOwner` for instance.

Reply:
Thanks you are completely right - I have made the suggested changes